### PR TITLE
Buttons are no longer indestructible

### LINKED ISSUE #40523: Tesla shocks are causing buttons to explode without being damaged or deleted, hence exploding repeatedly
Issue reported from  Round ID: 94497 (/tg/Station Sybil [ENGLISH] [US-EAST] [100% LAG FREE])  
Reporting client version: 512  
Testmerges: #40480: 'Makes the orbiter component less prone to failure' by ninjanomnom at commit 207a3044a0

### LINKED ISSUE #40751: Indestructible structures keep exploding by tesla
[Round ID]: # (If you discovered this issue from playing tgstation hosted servers:)
[Round ID]: # (**INCLUDE THE ROUND ID**)
[Round ID]: # (It can be found in the Status panel or retrieved from https://atlantaned.space/statbus/round.php ! The round id let's us look up valuable information and logs for the round the bug happened.)

[Testmerges]: # (If you believe the issue to be caused by a test merge [OOC tab -> Show Server Revision], report it in the pull request's comment section instead.)

[Reproduction]: # (Explain your issue in detail, including the steps to reproduce it. Issues without proper reproduction steps or explanation are open to being ignored/closed by maintainers.)

[For Admins]: # (Oddities induced by var-edits and other admin tools are not necessarily bugs. Verify that your issues occur under regular circumstances before reporting them.)

Indestructible structures such as buttons do not get deleted by explosions caused by Tesla, causing continuous and laggy explosions in that area. 

I might suggest making buttons not indestructible or having indestructible items not be affected by Tesla.

### PR DESCRIPTION
**Pull request by 81Denton**
:cl: Denton
fix: Buttons are no longer indestructible and won't cause endless tesla zap explosions. Buttons on CentCom and other special areas are still unbreakable.
/:cl:

Fixes: #40751
Fixes: #40523

### CODE REVIEW

**Comment by Ratvar** [comment]
Finally I can esword morgue tray button instead of needing a screw!

**Comment by Anonmare** [comment]
Do they really need to be acid-proof/unacidable?

Also we really need a way to make new buttons

**Comment by ATH1909** [comment]
+1 to the "way to make new buttons" idea.

**Review by ShizCalev — CHANGES_REQUESTED** [review]
Add /indestructible typepaths and change out all the buttons on CentCom.dmm, snowdin_vr.dmm, and syndicate_trainer.dmm

File: code/game/machinery/buttons.dm [hunk 1]
```diff
@@ -13,7 +13,7 @@
 	armor = list("melee" = 50, "bullet" = 50, "laser" = 50, "energy" = 50, "bomb" = 10, "bio" = 100, "rad" = 100, "fire" = 90, "acid" = 70)
 	use_power = IDLE_POWER_USE
 	idle_power_usage = 2
-	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	resistance_flags = LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
```
Comment (ShizCalev) [inline code/game/machinery/buttons.dm]: can probably remove UNACIDABLE too.

**Comment by PKPenguin321** [comment]
by the way, an alternate fix would be overriding tesla_act to force it to delete ([like with the gravity generator](https://github.com/tgstation/tgstation/blob/master/code/modules/power/gravitygenerator.dm#L42)) buuuuut this fix is probably better since buttons shouldnt be indestrutible to begin with

File: code/game/machinery/buttons.dm (hunk 1 above)
Comment (81Denton) [reply code/game/machinery/buttons.dm]: Is LAVA_PROOF still neccessary, or is that for buttons that are placed on lava tiles?

**Comment by 81Denton** [comment]
@PKPenguin321 Yeah, the proper fix would be making them breakable and rebuildable like most machinery. Just takes a bunch of work, you'd need to give some objs like crematoriums pulsable wires too.

**Review by ShizCalev — APPROVED** [review]

**Pull request merged**